File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/audit/InMemoryAiAuditLogRepository.java
package com.cleverferret.v2.feature.ai.audit;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public final class InMemoryAiAuditLogRepository implements AiAuditLogRepository {
    private final List<AiAuditEvent> events = new ArrayList<>();

    @Override
    public void append(AiAuditEvent event) {
        events.add(event);
    }

    public List<AiAuditEvent> events() {
        return Collections.unmodifiableList(events);
    }
}

File: CleverFerretV2/feature/ai/src/main/java/com/cleverferret/v2/feature/ai/audit/InMemoryAiAuditLogRepository.kt
package com.cleverferret.v2.feature.ai.audit

class InMemoryAiAuditLogRepository : AiAuditLogRepository {
    private val _events = mutableListOf<AiAuditEvent>()
    override fun append(event: AiAuditEvent) { _events += event }
    fun events(): List<AiAuditEvent> = _events.toList()
}
